lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -1785,14 +1790,22 @@ void ModuleEmitter::emitStatement(rtl::InstanceOp op) {
   SmallVector<rtl::ModulePortInfo, 8> portInfo;
   getModulePortInfo(moduleOp, portInfo);
 
+  // Emit the argument and result ports.
   auto opArgs = op.inputs();
   auto opResults = op.getResults();
   for (auto &elt : portInfo) {
+    // Figure out which value we are emitting.
     bool isLast = &elt == &portInfo.back();
-    StringRef valueName = elt.isOutput() ? getName(opResults[elt.argNum])
-                                         : getName(opArgs[elt.argNum]);
-    indent() << "  ." << StringRef(elt.getName()) << " (" << valueName
-             << (isLast ? ")\n" : "),\n");
+    Value portVal = elt.isOutput() ? opResults[elt.argNum] : opArgs[elt.argNum];
+
+    // Emit the port's name.
+    indent() << "  ." << StringRef(elt.getName()) << " (";
+
+    // Emit the value as an expression.
+    ops.clear();
+    emitExpression(portVal, ops);
```

Comment:
This seems very reasonable to me

Reply:
I don't follow the logic exactly here.  If it has no name, is this just emitting a dot?  Can you add a testcase that shows the behavior here?

I think that more generally we have two big paths we can go down:

1) Forbid unnamed ports in the verifier.
2) Have the asmprinter generate anonymous names

I would prefer #1 if we can get away with it.

Reply:
The part that comes [between the parentheses](https://github.com/llvm/circt/pull/302/files#diff-ae4df103f90fb51f0504ab56325762731b5a0afa45622ffcadf51da8bf596aa3R32) in these tests is all that changed here. Also note that we re-use a modport in that test and it is emitted with the same name both times.

The difference in the logic basically boils down to always using `emitExpression` instead of always using `getName`.

The value here is either defined by a wire or a `sv.modport.get`, which is verified to refer to an instance of a declared interface, as well as a real modport in that interface. The implementation of `emitExpression` will emit this operation inline, or else look up the name:

https://github.com/llvm/circt/blob/aef375e78fd7eee3ccc72e5796afce84a511c392/lib/EmitVerilog/EmitVerilog.cpp#L1396-L1399

I think we are already doing option 1. unless I misunderstand something. The `sv.modport.get` values have the needed verification, and otherwise we generate wires for every non-inline expression that comes through `collectNamesEmitDecls`. Is the concern about other inline expressions being used in an instance's port that might not emit as identifiers?

Reply:
@lattner thanks for the review. I'm going to land this as-is, but I also want to make sure your concerns here get addressed. Please take a look whenever you get around to it and let me know if the above explanation makes sense, or if we should handle this differently. I can add a follow-up PR to address any further concerns if necessary. Happy holidays.